.github/workflows/e2e-tests.yaml

```
@@ -71,3 +90,27 @@ jobs:
+
+  kuberay-client-tests:
+    runs-on: ubuntu-latest
+    needs: setup-cluster
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)